src/ring_theory/derivation.lean

```diff
@@ -540,4 +541,171 @@ begin
     exact submodule.subset_span }
 end
 
+/--
+The module of Kähler differentials (Kahler differentials, Kaehler differentials).
+This is implemented as `I / I ^ 2` with `I` the kernel of the multiplication map `S ⊗[R] S →ₐ[R] S`.
+To view elements as a linear combination of the form `s • D s'`, use
+`derivation_module.tensor_product_to_surjective` and `derivation.tensor_product_to_tmul`.
+
+We also provide the notation `Ω[S⁄R]` for `derivation_module R S`.
+Note that the slash is `\textfractionsolidus`.
+-/
+@[derive [add_comm_group, module R, module S, module (S ⊗[R] S)]]
+def derivation_module : Type* := (derivation_module.ideal R S).cotangent
```

Comment:
Hmm, should this be called `kaehler_differential`?

Whatever we end up choosing, this defn deserves a mention in the module docstring at the top of the file.